src/components/structures/RoomDirectory.js

```diff
@@ -497,6 +497,7 @@ export default class RoomDirectory extends React.Component {
         }
 
         let topic = room.topic || '';
+        // additional truncation based on line numbers is done via CSS
```

Comment:
extending this comment to explain why we also truncate the raw string (because we want to avoid a ridiculously long string being put into the DOM) and then its perfect!

Reply:
:heavy_check_mark: Will do

Reply:
Done:
```diff
$ git diff @{u} 
diff --git a/src/components/structures/RoomDirectory.js b/src/components/structures/RoomDirectory.js
index c89303859..df580e8de 100644
--- a/src/components/structures/RoomDirectory.js
+++ b/src/components/structures/RoomDirectory.js
@@ -497,7 +497,9 @@ export default class RoomDirectory extends React.Component {
         }
 
         let topic = room.topic || '';
-        // additional truncation based on line numbers is done via CSS
+        // Additional truncation based on line numbers is done via CSS,
+        // but to ensure that the DOM is not polluted with a huge string
+        // we give it a hard limit before rendering.
         if (topic.length > MAX_TOPIC_LENGTH) {
             topic = `${topic.substring(0, MAX_TOPIC_LENGTH)}...`;
         }
```